### PR TITLE
Add -XX:[+|-]PortableSharedCache

### PR DESCRIPTION
**Pull request by harryyu1994**
Issue: https://github.com/eclipse/openj9-docs/issues/615

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
On hold till https://github.com/eclipse/openj9/pull/9731 is merged.

**Comment by harryyu1994** [comment]
But I guess it's open for review. FYI @mpirvu @hangshao0

File: docs/xshareclasses.md [hunk 1]
```diff
@@ -588,6 +588,12 @@ behavior, which can improve the performance of class loading from the shared cla
 
 : Gives detailed output on the cache I/O activity, listing information about classes that are stored and found. Each class loader is given a unique ID (the bootstrap loader is always 0) and the output shows the class loader hierarchy at work, where class loaders must ask their parents for a class before they can load it themselves. It is typical to see many failed requests; this behavior is expected for the class loader hierarchy.
 
+### `portable`
+
+        -Xshareclasses:portable
+
+:   Enables AOT code to be compiled based on a chosen set of processor features that maximizes its portability across machines. It is currently only supported on x86.
```
Comment (mpirvu) [inline docs/xshareclasses.md:595]: I am not a native English speaker, but "It is currently supported only on x86" sounds better to me.
Also, is support enabled for 32-bit as well? I guess so, but if not we may want to qualify further.
Also, maybe it's a good idea to specify the chosen processor features (features present in the Intel Sandy Bridge processors)

**Comment by DanHeidinga** [comment]
See discussion in https://github.com/eclipse/openj9/pull/9731#issuecomment-648333667 - let's go with just the `-XX:+-` option and remove the `-Xshareclasses:portable` suboption

File: docs/xshareclasses.md (hunk 1 above)
Comment (harryyu1994) [reply docs/xshareclasses.md:595]: Do you think we should spell out all the features or just state "features present in Intel Sandy Bridge"?

Comment (mpirvu) [reply docs/xshareclasses.md:595]: I think users will be more familiar with the SandyBridge name than with each individual feature. It also gives them a simple reference point.

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
@harryyu1994  Please confirm that this PR is not complete and should **not** be merged for the 0.21.0 release. Thanks!

**Comment by mpirvu** [comment]
@pmhayward Indeed, this PR is not going into 0.21.0 and hence no new options will be added to that release.

**Comment by harryyu1994** [comment]
Ready for review.

**Comment by pmhayward** [comment]
Jenkins doc stage

File: docs/xxportablesharedcache.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:\[+|-\]PortableSharedCache
+
+This feature enables AOT compiled code to be generated based on a chosen set of processor features (features present in the Intel SandyBridge processors) that maximizes its portability across machines. It is currently supported only on x86. The feature is turned on by default in Docker containers that can be disabled with "-XX:-PortableSharedCache". The portable shared cache feature state is persisted in the SCC which means specifying it with an existing SCC has no effect.
```
Comment (pmhayward) [inline docs/xxportablesharedcache.md:27]: Intel SandyBridge -- Is this now Intel Celeron? (Or am I just confused!) Trademarks required:  Intel&reg; Celeron&reg; (I don't think SandyBridge would need one.)
`-XX:-PortableSharedCache` -- pleas make monospace with back ticks as in table.
SCC - on first occurrence, please spell out: "...shared classes cache (SCC)..."

Comment (harryyu1994) [reply docs/xxportablesharedcache.md]: Actually I think it should be Intel Sandy Bridge (https://en.wikipedia.org/wiki/Sandy_Bridge).

**Comment by harryyu1994** [comment]
> Also, just for the record and for future reference, would appreciate an issue linked to this PR, that gives basic gist of documentation required for this new feature. Thanks! (P.S. Away now until 5 Aug)

Issue related (https://github.com/eclipse/openj9-docs/issues/615). Ready for review again.

File: docs/version0.22.md [hunk 3]
```diff
@@ -44,6 +45,9 @@ The latest builds of OpenJDK with OpenJ9 for Java 8 and 11 at the AdoptOpenJDK c
 
 To learn more about support for OpenJ9 releases, including OpenJDK levels and platform support, see [Supported environments](openj9_support.md).
 
+### New `-XX:[+|-]PortableSharedCache` option added
+
+This option enables AOT compiled code to be generated based on a chosen set of processor features that maximizes its portability across machines. It is currently supported only on x86. The feature is turned on by default in Docker containers that can be disabled with `-XX:-PortableSharedCache`. For more information, see [`-XX:[+|-]PortableSharedCache`](xxportablesharedcache.md).
```
Comment (mpirvu) [inline docs/version0.22.md:50]: > `that can be disabled with -XX:-PortableSharedCache`  

I would replace `that` with `but` or maybe with `and`.

File: docs/xxportablesharedcache.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -XX:\[+|-\]PortableSharedCache
+
+This option enables AOT compiled code to be generated based on a chosen set of processor features (features present in the Intel Sandy Bridge microarchitecture) that maximizes its portability across machines. It is currently supported only on x86. The feature is turned on by default in Docker containers that can be disabled with `-XX:-PortableSharedCache`. To enable the portable shared cache feature outside Docker containers, `-XX:+PortableSharedCache` needs to be specified for the initial Java Virtual Machine instance (when the creation of the Shared Class Cache happens) as well as for every subsequent instances that make use of the same Shared Class Cache.
```
Comment (mpirvu) [inline docs/xxportablesharedcache.md:27]: > as well as for every subsequent instances that make use of the same Shared Class Cache.

I would use singular for here: `as well as for every subsequent instance that makes use of the same Shared Class Cache.`

File: docs/version0.22.md (hunk 3 above)
Comment (SueChaplain) [reply docs/version0.22.md:50]: Peter is out until Wednesday. Please replace "that" with "and".

File: docs/xxportablesharedcache.md (hunk 4 above)
Comment (SueChaplain) [reply docs/xxportablesharedcache.md:27]: Agree with Marius, please update.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by harryyu1994** [comment]
Ready for review

**Comment by pmhayward** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
@harryyu1994 Thanks for making the updates!

**Pull request merged**